### ISSUE TITLE
Spearhead: using incorrect Headstart annotations on Theme Switch

### ISSUE THREAD

**Issue by ianstewart** [Automattic/wp-calypso#52799]
When we activate Spearhead in the theme showcase and choose the option that tells Dotcom to set up the site to look just like the theme homepage we get the following result.

![image](https://user-images.githubusercontent.com/545779/118031055-b5344780-b32b-11eb-9c9c-3942facc1670.png)

Expected result

![image](https://user-images.githubusercontent.com/545779/118031173-d85ef700-b32b-11eb-8e70-c1db5fc97030.png)

**Comment by michaeldcain** [comment]
It looks like the annotation already exists: D61318-code. I'm thinking the theme doesn't have the `auto-loading-homepage` tag

**Comment by michaeldcain** [comment]
It does have the tag.

**Comment by michaeldcain** [comment]
So I did a little more looking. The issue here is that Headstart on theme_switch doesn't account for themes where `"show_on_front": "posts"`. The Spearhead demo site actually has a Home page being created that matches your screenshot above, it's just that the templating system skips that page, since the theme is a blog.

**Comment by michaeldcain** [comment]
Adding to the complication, Headstart on switch_theme only creates a single page (the homepage), so if we fix it to instead update the `show_on_front` setting, and the user doesn't have any posts, they'll be shown the empty blog home template.

**Comment by ianstewart** [comment]
> we fix it to instead update the show_on_front setting, and the user doesn't have any posts, they'll be shown the empty blog home template

The current status is much worse than this. :)

> the empty blog home template

![image](https://user-images.githubusercontent.com/545779/118034620-ced78e00-b32f-11eb-891b-8891345084f5.png)

I think this is better than the current status in that case.

Perhaps later it could be improved by headstarting one or two sample posts. (The grey-boxed part of the design is a sticky post.) But then I'd want to introduce removal of Headstarted posts on theme switch and I think that complicates a shippable fix to start.

**Comment by ianstewart** [comment]
> The Spearhead demo site actually has a Home page being created that matches your screenshot above

**Note:** I deleted this from the demo a few minutes ago. Seems like an oversight.

**Comment by ianstewart** [comment]
> account for themes where "show_on_front": "posts"

How difficult is this?

**Comment by jordesign** [comment]
> we fix it to instead update the show_on_front setting, and the user doesn't have any posts, they'll be shown the empty blog home template

As a separate issue - I've long believed the 'Nothing Found' text for no posts could be a lot better. Is this something on a theme-by-theme basis (I'm guessing so).

**Comment by ianstewart** [comment]
> As a separate issue - I've long believed the 'Nothing Found' text for no posts could be a lot better. Is this something on a theme-by-theme basis (I'm guessing so).

Comes from a template that we generally carry over from theme to theme. Would be fun to improve it a bit for the cases where it still pops up. (And make sure that when we start using Query Block variations that we improve it there as well!)

**Comment by jordesign** [comment]
> Would be fun to improve it a bit for the cases where it still pops up.

Have opened this is a separate issue.
https://github.com/Automattic/wp-calypso/issues/52806

**Comment by michaeldcain** [comment]
D61318-code has been deployed. That really only ensures that the correct content is created on Signup or with the RC tool, not theme switch (which requires the theme switch tool to handle blogs).

**Comment by ianstewart** [comment]
How might we account for themes where `"show_on_front": "posts"` in the code that runs the template set up on theme switch?

**Comment by michaeldcain** [comment]
If we approach it with a theme tag (i.e. `blog-homepage`), then D61382-code would handle setting `"show_on_front": "posts"`.

**Comment by michaeldcain** [comment]
@blackjackkent, just some notes if you pursue the `blog-homepage` theme tag route:

- D61382-code would handle the backend changes to set the correct option on theme switch. It's approved now - I'm happy to merge it.
- Any themes we want to support (#52757) will need the `blog-homepage` theme tag (i.e. D33616-code) and will need their annotations double-checked to make sure we're not Headstarting unused homepages (i.e. D61318-code).
- We'll need to decide what we'd want to do with the `auto-loading-homepage` tag in these same themes. That tag might be used for some templating logic (ref: `full-site-editing/class-template-first-themes.php?r=215775#303`) but if it's not, it's still required to get the Calypso modal to show on the theme showcase (see [`themeHasAutoLoadingHomepage` selector](https://github.com/Automattic/wp-calypso/blob/38c37bbfe35b8c741c4e346637987ddc0fb3ab59/client/state/themes/selectors/theme-has-auto-loading-homepage.js)). If you decide to remove it from the blog-homepage theme tags, you'll need to update the logic that show that modal option.

Let me know if I can help! /cc @ianstewart @sixhours

**Comment by sixhours** [comment]
It looks like what's left here is to:

* [x] Deploy D61382-code
* Add `blog-homepage` and `auto-loading-homepage` tags to themes' stylesheets from the initial list:
* [x] Dyad 2 - D61635-code
* [x] Spearhead - D61383-code
* [x] Independent Publisher 2 - D61602-code
* [x] Hemingway Rewritten - D61624-code
* Make sure we're not adding unnecessary home pages to blog homepage annotations:
* [x] Dyad 2 - D61635-code
* [x] Spearhead - D61383-code
* [x] Independent Publisher 2 - D61602-code
* [x] Hemingway Rewritten - D61624-code
* [x] Deploy D61673-code so users switching from static front page to blog homepage themes have their home pages set to Draft
* [x] Update language in the theme switch modal to better reflect both blog homepage themes and static front page themes - #53082
* [ ] When we're ready, add the `Auto Loading Homepage` tag to any themes in the Theme Showcase that we want to move to the "Recommended themes" section.
* [x] Submit new annotations for translation (this isn't a blocker IMO)

**Comment by michaeldcain** [comment]
> Update logic around auto-loading-homepage for blog-homepage themes, maybe. I'm still a bit fuzzy and need to research how that tag affects things.

I think that one glaring issue is the way the theme switch behavior for blogs is presented to the user: "Use Spearhead's homepage content and make my existing homepage a draft." This isn't technically true. We don't add blog posts or draft their current homepage. Adding a `blog-homepage` selector to Calypso would let us clean up that message.

**Comment by michaeldcain** [comment]
> Deploy D61382-code

✅

**Comment by sixhours** [comment]
> We don't add blog posts or draft their current homepage.

Ahhh, is this an issue? I could see it being problematic if they lose everything because they chose a different type of front page setting. 😬 Or does the existing homepage get preserved as a regular published page instead of a draft?

**Comment by michaeldcain** [comment]
The existing homepage gets preserved as a regular published page. I could try to create a patch to handle, at minimum, drafting it.

**Comment by ianstewart** [comment]
> "Use Spearhead's homepage content and make my existing homepage a draft." This isn't technically true. We don't add blog posts or draft their current homepage. Adding a blog-homepage selector to Calypso would let us clean up that message.

Perhaps we could fix this language up for everyone.

> Use Spearhead's home page for my site and draft my current home page.

More accurate in this instance with no mention of "content"? Perhaps more understandable for all themes?

> The existing homepage gets preserved as a regular published page. I could try to create a patch to handle, at minimum, drafting it.

This should be a blocker so the experience is the same across the board. 

Both of the following flows should work as expected with that change as well.

1. Start my site with Mayland.
2. Switch to Spearhead and select make my homepage look like Spearhead and draft the current one.
3. Switch to Mayland and select keep my current homepage. (Should look like the "blog"/index home page of Mayland.)

1. Start my site with Mayland.
2. Switch to Spearhead and select make my homepage look like Spearhead and draft the current one.
3. Switch to Mayland and select use the Mayland homepage. (Should look like the cool Mayland gallery page.)

**Comment by michaeldcain** [comment]
> This should be a blocker so the experience is the same across the board.

Here's a patch to draft the previous homepage: D61673-code

**Comment by sixhours** [comment]
I applied `auto-loading-homepage` to the themes in the short list above since it looks like that's the direction we're headed.

**Comment by jordesign** [comment]
> More accurate in this instance with no mention of "content"? Perhaps more understandable for all themes?

I personally prefer this - in terms of phrasing that works for showing either posts *or* page content on the home page.

And agree that it should be consistent behaviour to the user - regardless of what is changing behind the scenes. Thanks folks!

**Comment by sixhours** [comment]
@mreishus and I did some quick testing this morning: p1621608042131700-slack-C0202PMUXLJ

TL;DR: 

* I deployed Hemingway Rewritten's annotations and stylesheet.
* Adding the `Auto Loading Homepage` tag to the themes in the Showcase itself is what triggers the addition of the theme to the Recommended section, so we should be fine to deploy the other annotations without adding anything to the Recommended section until we're ready.
* It looks like there are two outstanding issues to ship before we should add the older themes to the Recommended list. I've added them to my list [to keep track of progress here](https://github.com/Automattic/wp-calypso/issues/52799#issuecomment-845228867).

**Comment by mreishus** [comment]
I'm seeing that switching to themes can move you from `show_on_front: "pages"` -> `show_on_front: "posts"`, but cannot make the opposite change.

I started with my site on Seedlet, showing static content on the homepages. I can switch between Seedlet and Twenty Twenty One, selecting to to replace my homepage each time. When I do, my homepage's static content gets updated to that theme's:

**Step 1: Seedlet (pages)**
![2021-05-21_16-08](https://user-images.githubusercontent.com/937354/119198869-b2360700-ba4f-11eb-8e7b-73691318677b.png)

Switch to 2021 using homepage replacement (All future switches will also use this):
![2021-05-21_16-14](https://user-images.githubusercontent.com/937354/119198897-ba8e4200-ba4f-11eb-9411-11c4171e872f.png)

**Step 1: Seedlet (pages) -> Step 2: 2021 (pages)**
![2021-05-21_16-09_1](https://user-images.githubusercontent.com/937354/119198937-cd087b80-ba4f-11eb-9e31-99cb8c54ddbd.png)

Now if I switch to Spearhead, and elect to replace my homepage, I'm switched to a list of posts to match the spearhead theme:

**Step 1: Seedlet (pages) -> Step 2: 2021 (pages) -> Step 3: Speadhead (posts)**
![2021-05-21_16-09_2](https://user-images.githubusercontent.com/937354/119198989-ead5e080-ba4f-11eb-93d2-5e98fcfabb6c.png)

Now I switch back to Seedlet and ask for my homepage to be replaced, but what I see is different:

**Step 1: Seedlet (pages) -> Step 2: 2021 (pages) -> Step 3: Speadhead (posts) -> Step 4: Seedlet (posts)**
![2021-05-21_16-10](https://user-images.githubusercontent.com/937354/119199086-148f0780-ba50-11eb-9bb0-16ed97f92c85.png)

Same with 2021:
**Step 1: Seedlet (pages) -> Step 2: 2021 (pages) -> Step 3: Speadhead (posts) -> Step 4: Seedlet (posts) -> Step 5: 2021 (posts)**
![2021-05-21_16-11](https://user-images.githubusercontent.com/937354/119199107-207ac980-ba50-11eb-934a-6b64b4ffb49b.png)

To sum up, what I see on Seedlet and 2021 is now different only because I have selected Spearhead in the past.

**Comment by mreishus** [comment]
Will the tag `blog-homepage` have a 1-1 relationship with the setting `settings.options.show_on_front: posts`? If so, I'm worried about these getting out of sync in the future.

**Comment by michaeldcain** [comment]
> I'm seeing that switching to themes can move you from show_on_front: "pages" -> show_on_front: "posts", but cannot make the opposite change.

@mreishus from what I can tell, this was an intentional decision in the original feature. If a `page_for_posts` option doesn't exist, they didn't want the user to "lose" their blog feed, so the code creates the new theme's homepage, but never sets it as the homepage. In D61673-code, ~L143 I've changed that to create a new page titled "Blog", set it as the `page_for_posts`, and then set the new TFT template as the homepage.

> Will the tag blog-homepage have a 1-1 relationship with the setting settings.options.show_on_front: posts? If so, I'm worried about these getting out of sync in the future.

I left a comment in Slack (p1621638223163900/1621630918.153700-slack-C0202PMUXLJ), but these aren't meant to be exactly 1:1. Every theme with the `blog-homepage` tag should have `settings.options.show_on_front: posts`, but not every theme with `settings.options.show_on_front: posts` should have the `blog-homepage` tag (only the ones we want this theme switch functionality exposed to for the time being). Similar exists for the `auto-loading-homepage` tag and themes with `settings.options.show_on_front: page` - there are lots of themes that we don't necessarily want to expose the theme switch behavior for do to complex theme settings or multi-page content requirements (b/c it's not technically a full Headstart).

**Comment by mreishus** [comment]
>Every theme with the blog-homepage tag should have settings.options.show_on_front: posts, but not every theme with settings.options.show_on_front: posts should have the blog-homepage tag.

Noting that this allows a configuration with undefined behavior to be expressed in the system: `show_on_front: pages` with `blog-homepage` tag.

**Comment by mreishus** [comment]
Followup issues:

- TwentyTwenty -> Spearhead (their menu still has 'blog' linking to nowhere)
- Spearhead -> TwentyTwenty (should Add a 'blog' item to their menu)
- "'Blog', 'Home'" menu, order changed
- Helper text next to "Home" on Calypso Pages view not always working (example: "Seedlet Theme")

**Comment by michaeldcain** [comment]
> TwentyTwenty -> Spearhead (their menu still has 'blog' linking to nowhere)
> Spearhead -> TwentyTwenty (should Add a 'blog' item to their menu)
> "'Blog', 'Home'" menu, order changed

I started these in D62000-code

**Comment by sixhours** [comment]
This last piece is part of an upcoming sprint: ` When we're ready, add the Auto Loading Homepage tag to any themes in the Theme Showcase that we want to move to the "Recommended themes" section.`

So I'm going to close this out and call it done. Thanks all for your help!
